tests/functional/docs/test_iam.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from tests.functional.docs import BaseDocsFunctionalTest
+from botocore.docs.service import ServiceDocumenter
+
+
+class TestIAMDocs(BaseDocsFunctionalTest):
+    def test_changes_policy_doc_to_dict(self):
+        modified_methods = ['create_instance_profile',
+                            'create_policy',
+                            'create_policy_version',
+                            'create_role',
+                            'get_account_authorization_details',
+                            'get_group_policy',
+                            'get_instance_profile',
+                            'get_policy_version',
+                            'get_role',
+                            'get_role_policy',
+                            'get_user_policy',
+                            'list_instance_profiles',
+                            'list_instance_profiles_for_role',
+                            'list_policy_versions',
+                            'list_roles',
+                            'put_group_policy',
+                            'put_role_policy',
+                            'put_user_policy',
+                            'update_assume_role_policy']
+        service_contents = ServiceDocumenter('iam', self._session)\
+            .document_service()
+        for method_name in modified_methods:
```

Comment:
It may be worth breaking this test out to rely on yield for when this test is ran. If this test ever fails, it will be hard to determine where exactly in the test it failed because it is just a big for loop. By using yield it will create a bunch of tests that will be easier to track down and if one operation fails, all of the operations will still be checked.

Otherwise, I would also be fine with just checking just two arbitrary operations (one for the input and one for the output). Checking all of the operations seem a little overkill because they all same the same logic especially since we now have unit test for the customization. Thoughts?